blockbuilder/genesisState.go

```diff
@@ -0,0 +1,50 @@
+package blockbuilder
+
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/NethermindEth/juno/core"
+	"github.com/NethermindEth/juno/core/felt"
+)
+
+var ErrChainIDRequired = errors.New("ChainID is required")
+
+type GenesisConfig struct {
+	ChainID                 string           `json:"chain_id"`
+	WhitelistedSequencerSet []felt.Felt      `json:"whitelisted_sequencer_set"`
+	State                   genesisStateData `json:"state"`
+}
+
+type genesisStateData struct {
+	Classes       map[string]genesisClassData    `json:"classes"`
+	Contracts     map[string]genesisContractData `json:"contracts"`
```

Comment:
```suggestion
	Classes       map[felt.Felt]genesisClassData    `json:"classes"`
	Contracts     map[felt.Felt]genesisContractData `json:"contracts"`
```

Reply:
Un-marshalling a map[felt.Felt]struct is giving errors, so we're just going to define the type as a []string, and calculate the classHash ourselves.